### PR TITLE
Agribalyse 20 oct

### PR DESCRIPTION
**Pull request by teolemon**

### CODE REVIEW

**Comment by teolemon** [comment]
Errors in the categories taxonomy definition:
ERROR - fr:blette already is a synonym of fr:blettes for entry en:chards - fr:bette cannot be mapped to entry en:swiss-chard
ERROR - fr:petits-pois-en-conserve already is a synonym of fr:petits-pois-appertises-egouttes for entry en:canned-garden-peas - fr:petits-pois-en-conserve cannot be mapped to entry en:canned-peas
Errors in the categories taxonomy definition at lib/ProductOpener/Tags.pm line 992.

**Comment by teolemon** [comment]
Errors in the categories taxonomy definition:
ERROR - en:cooked-sweet-green-peppers has an undefined parent 
ERROR - en:cooked-sweet-red-peppers has an undefined parent

File: taxonomies/categories.txt [hunk 1]
```diff
@@ -22190,6 +22192,16 @@ agribalyse_food_code:en:4016
 ciqual_food_code:en:4016
 ciqual_food_name:en:Potato flakes, dehydrated, with milk or cream
 ciqual_food_name:fr:Pomme de terre, flocons déshydratés, au lait ou à la crème
+ 
+TODO: dehydrated to be
```
Comment (teolemon) [inline taxonomies/categories.txt:22196]:
```suggestion
TODO: dehydrated to be added
```

File: taxonomies/categories.txt [hunk 2]
```diff
@@ -51758,23 +51855,23 @@ ciqual_food_code:en:25033
 ciqual_food_name:en:Burgundy-style beef stew
 ciqual_food_name:fr:Boeuf bourguignon
 
-<en:Beef dishes
+<en:Bœufs bourguignons
```
Comment (stephanegigandet) [inline taxonomies/categories.txt]: There are 2 different things:
- "Boeufs bourguignons" (the prepared dish, with meat and vegetables)
- and "Beef stewing meat" ("Boeuf, à bourguignon ou pot-au-feu).

en:Boeufs bourguignons is under en:beef stews, and it should not be a parent of cooked or raw beef stew meat .

There are 2 AGB categories:

6231	6231	viandes, œufs, poissons	viandes crues	Boeuf, à bourguignon ou pot-au-feu, cru	Beef, stewing meat, raw
25033	25033	entrées et plats composés	plats composés	Boeuf bourguignon	Burgundy-style beef stew
11120	11120	aides culinaires et ingrédients divers	sauces	Sauce bourguignonne, préemballée	Burgundy-style sauce, prepacked

6231 | 6231 | viandes, œufs, poissons | viandes crues | Boeuf, à bourguignon ou pot-au-feu, cru | Beef, stewing meat, raw
-- | -- | -- | -- | -- | --
25033 | 25033 | entrées et plats composés | plats composés | Boeuf bourguignon | Burgundy-style beef stew
11120 | 11120 | aides culinaires et ingrédients divers | sauces | Sauce bourguignonne, préemballée | Burgundy-style sauce, prepacked



25033 | 25033 | entrées et plats composés | plats composés | Boeuf bourguignon | Burgundy-style beef stew
-- | -- | -- | -- | -- | --

File: taxonomies/categories.txt [hunk 3]
```diff
@@ -53144,6 +53219,40 @@ en:Cassoulets from Toulouse
 fr:Cassoulets toulousains
 nl:Cassoulet uit Toulouse
 
+<en:Duck dishes
+<en:Cassoulets
+en:Cassoulets with duck confit
+fr:Cassoulets au confit de canard
+nl:Cassoulet met geconfijte eend
+agribalyse_proxy_food_code:en:25099
+agribalyse_proxy_food_name:en:White bean stew, with duck or goose, canned
```
Comment (stephanegigandet) [inline taxonomies/categories.txt:53228]: why agribalyse_proxy instead of just agribalyse?

Comment (stephanegigandet) [reply taxonomies/categories.txt:53228]: if it's to make a distinction between cassoulet with duck and cassoulet with duck confit, then we should just merge them. The duck is always duck confit in cassoulet.

File: taxonomies/categories.txt [hunk 4]
```diff
@@ -62334,6 +62450,14 @@ it:Pasta senza glutine
 nl:Glutenvrije pastas
 pl:Makaron bezglutenowy
 
+<en:Gluten-free pasta
+<en:Dry pastas
+en:Gluten-free dry pasta, Dry gluten-free pasta, Dried gluten-free pasta
+fr:Pâtes sèches sans gluten
+agribalyse_proxy_food_code:en:9824
+agribalyse_proxy_food_name:en:Dried pasta, gluten-free, cooked, unsalted
```
Comment (stephanegigandet) [inline taxonomies/categories.txt:62458]: same here, it's not really a proxy, it's the actual match

File: taxonomies/categories.txt [hunk 5]
```diff
@@ -71341,6 +71412,13 @@ la:Brassica oleracea var. capitata
 nl:Kolen
 
 <en:Cabbages
+en:White cabbage
+fr:Chou blanc
+agribalyse_proxy_food_code:en:20116
+agribalyse_proxy_food_name:en:White cabbage, raw
```
Comment (stephanegigandet) [inline taxonomies/categories.txt:71418]: exact match

**Review by stephanegigandet — CHANGES_REQUESTED** [review]
Added a few comments.

File: taxonomies/categories.txt (hunk 2 above)
Comment (teolemon) [reply taxonomies/categories.txt]: <en:Beef
en:Beef stew meat
fr:Boeuf à bourguignon, Boeuf à Pot-au-feu
agribalyse_food_code:en:6231
ciqual_food_code:en:6231
ciqual_food_name:en:Beef, stewing meat, raw
ciqual_food_name:fr:Boeuf, à bourguignon ou pot-au-feu, cru

Comment (teolemon) [reply taxonomies/categories.txt]: Sauce bourguignonne was already OK. Nothing to report.

Comment (teolemon) [reply taxonomies/categories.txt]: and Beef bourguignon is kept as a stew.

File: taxonomies/categories.txt (hunk 3 above)
Comment (teolemon) [reply taxonomies/categories.txt]: merged

File: taxonomies/categories.txt (hunk 4 above)
Comment (teolemon) [reply taxonomies/categories.txt]: fixed

File: taxonomies/categories.txt (hunk 5 above)
Comment (teolemon) [reply taxonomies/categories.txt]: fixed

**Comment by teolemon** [comment]
Errors in the categories taxonomy definition:
ERROR - en:canned-soy-sprouts is a parent of itself
ERROR - en:frozen-corn-on-the-cob has an undefined parent 
Errors in the categories taxonomy definition at lib/ProductOpener/Tags.pm line 1634.
[categories] child process exited with code 25

**Review by stephanegigandet — APPROVED** [review]

**Pull request merged**